cirq-core/cirq/transformers/routing/mapping_manager.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, TYPE_CHECKING
+import networkx as nx
+
+from cirq import ops, protocols
+
+if TYPE_CHECKING:
+    import cirq
+
+
+class MappingManager:
+    """Class that keeps track of the mapping of logical to physical qubits and provides
+    convenience methods for distance queries on the physical qubits.
+
+    Qubit variables with the characer 'p' preppended to them are physical and qubits with  the
+    character 'l' preppended to them are logical qubits.
+    """
+
+    def __init__(self, device_graph: nx.Graph, initial_mapping: Dict[ops.Qid, ops.Qid]) -> None:
+        """Initializes MappingManager.
+
+        Args:
+            device_graph: connectivity graph of qubits in the hardware device.
+            circuit_graph: connectivity graph of the qubits in the input circuit.
+            initial_mapping: the initial mapping of logical (keys) to physical qubits (values).
+        """
+        self.device_graph = device_graph
+        self._map = initial_mapping.copy()
+        self._inverse_map = {v: k for k, v in self._map.items()}
+        self._induced_subgraph = nx.induced_subgraph(self.device_graph, self._map.values())
+        self._shortest_paths_matrix = dict(nx.all_pairs_shortest_path(self._induced_subgraph))
+
+    @property
+    def map(self) -> Dict[ops.Qid, ops.Qid]:
```

Comment:
For type annotations use `cirq.Qid` instead of `ops.Qid`; here and elsewhere (for all types, not just `Qid`). This works because you have a conditional `import cirq` based on whether the type checker is analyzing the file. 

This is useful because if we decide to move `Qid` to a different module, we won't have to update all the type annotations that depend on `ops.Qid`, since `cirq.Qid` would still be valid. 
```suggestion
    def map(self) -> Dict['cirq.Qid', 'cirq.Qid']:
```